src/webgpu/shader/validation/variable_and_const.spec.ts

```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
     const { storageClass, containerType, scalarType } = t.params;
     const type = containerType ? `${containerType}<${scalarType}>` : scalarType;
 
-    const code = `
-      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
+    var code;
+    if (`${storageClass}` === 'in') {
+      code = `
+        struct MyInputs {
+          [[location(0)]] a : ${type};
+        };
+
+        [[stage(vertex)]]
+        fn main(inputs : MyInputs) {
+        }
+      `;
+    } else if (`${storageClass}` === 'out') {
+      code = `
+        struct MyOutputs {
+          [[location(0)]] a : ${type};
+        };
+
         [[stage(vertex)]]
-        fn main() {
+        fn main() -> MyOutputs {
+          return MyOutputs();
         }
       `;
+    } else {
+      code = `
+      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
```

Comment:
Presumably this is just for the `private` storage class case, in which case the `[[location(0)]]` decoration is not valid here.

Reply:
Done